homeassistant/components/snapcast/__init__.py

```diff
@@ -1 +1,43 @@
-"""The snapcast component."""
+"""Snapcast Integration."""
+import logging
+
+import snapcast.control
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN, PLATFORMS
+from .server import HomeAssistantSnapcast
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Snapcast from a config entry."""
+    host = entry.data[CONF_HOST]
+    port = entry.data[CONF_PORT]
+    try:
+        server = await snapcast.control.create_server(
+            hass.loop, host, port, reconnect=True
+        )
+    except OSError as ex:
+        raise ConfigEntryNotReady(
+            f"Could not connect to Snapcast server at {host}:{port}"
+        ) from ex
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = HomeAssistantSnapcast(
+        server, [], []
+    )
```

Comment:
Could you add a default to the dataclass instead?
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = HomeAssistantSnapcast(
        server
    )
```